salt/modules/cryptdev.py

```diff
@@ -0,0 +1,344 @@
+    devices = __salt__['cmd.run_stdout']('dmsetup ls --target crypt')
+    out_regex = re.compile(r'(?P<devname>\w+)\W+\((?P<major>\d+), (?P<minor>\d+)\)')
+
+    log.debug(devices)
+    for line in devices.split('\n'):
+        match = out_regex.match(line)
+        if match:
+            dev_info = match.groupdict()
+            ret[dev_info['devname']] = dev_info
+        else:
+            log.warn('dmsetup output does not match expected format')
+
+    return ret
+
+
+def crypttab(config='/etc/crypttab'):
+    '''
+    List the contents of the crypttab
+
+    CLI Example:
+
+    .. code-block:: bash
+
+        salt '*' cryptdev.crypttab
+    '''
+    ret = {}
+    if not os.path.isfile(config):
+        return ret
+    with salt.utils.fopen(config) as ifile:
+        for line in ifile:
+            try:
+                entry = _crypttab_entry.dict_from_line(line)
+
+                entry['options'] = entry['options'].split(',')
+
+                # Handle duplicate names by appending `_`
+                while entry['name'] in ret:
+                    entry['name'] += '_'
+
+                ret[entry.pop('name')] = entry
+            except _crypttab_entry.ParseError:
+                pass
+
+    return ret
+
+
+def rm_crypttab(name, config='/etc/crypttab'):
+    '''
+    Remove the named mapping from the crypttab. If the described entry does not
+    exist, nothing is changed, but the command succeeds by returning
+    ``'absent'``. If a line is removed, it returns ``'change'``.
+
+    CLI Example:
+
+    .. code-block:: bash
+
+        salt '*' cryptdev.rm_crypttab foo
+    '''
+    modified = False
+    criteria = _crypttab_entry(name=name)
+
+    # For each line in the config that does not match the criteria, add it to
+    # the list. At the end, re-create the config from just those lines.
+    lines = []
+    try:
+        with salt.utils.fopen(config, 'r') as ifile:
+            for line in ifile:
+                try:
+                    if criteria.match(line):
+                        modified = True
+                    else:
+                        lines.append(line)
+
+                except _crypttab_entry.ParseError:
+                    lines.append(line)
+
+    except (IOError, OSError) as exc:
+        msg = "Couldn't read from {0}: {1}"
+        raise CommandExecutionError(msg.format(config, str(exc)))
+
+    if modified:
+        try:
+            with salt.utils.fopen(config, 'w+') as ofile:
+                ofile.writelines(lines)
+        except (IOError, OSError) as exc:
+            msg = "Couldn't write to {0}: {1}"
+            raise CommandExecutionError(msg.format(config, str(exc)))
+
+    # If we reach this point, the changes were successful
+    return 'change' if modified else 'absent'
+
+
+def set_crypttab(
+        name,
+        device,
+        password='none',
+        options='',
+        config='/etc/crypttab',
+        test=False,
+        match_on='name',
+        **kwargs):
```

Comment:
Thanks for your speedy review! I think I cargo-culted this accidentally from the `mount` module, on which most of this code was based. I'll remove it.